chapter04/GamblersProblem.py

```diff
@@ -20,9 +20,13 @@
 # optimal policy
 policy = np.zeros(GOAL + 1)
 
-# state value
+# reward Value
+rewardValue = np.zeros(GOAL + 1)
+rewardValue[GOAL] = 1.0
+
+# state Value
 stateValue = np.zeros(GOAL + 1)
-stateValue[GOAL] = 1.0
```

Comment:
Chap 4.5 is very concise and wasn't enough to get the idea of "asynchronous". Out code seems to sweep all the state and it give me an impression it's in-place but not asynchronous.
(I know the book says "Asynchronous DP algorithms are in-place interactive DP algorithm though I can't infer this means Asynchronous == In-place)

But the code computes correctly as you said. I'll cancel my pull request. 
Thanks for your reply. This gave me an good opportunity to think about the difference.

By the way, the following URL explains asynchronous value iteration, which I used to understand. I'll leave the URL hoping it might be helpful for someone who have same question as me.
http://www.cs.ubc.ca/~kevinlb/teaching/cs322%20-%202007-8/Lectures/DT5.pdf
 
Thanks.